.golangci.yml

```diff
@@ -0,0 +1,195 @@
+run:
+  deadline: 10m
+
+  skip-files:
+  - "zz_\\..+\\.go$"
+
+output:
+  # colored-line-number|line-number|json|tab|checkstyle|code-climate, default is "colored-line-number"
+  format: colored-line-number
+
+linters-settings:
+  errcheck:
+    # report about not checking of errors in type assetions: `a := b.(MyStruct)`;
+    # default is false: such cases aren't reported by default.
+    check-type-assertions: false
+
+    # report about assignment of errors to blank identifier: `num, _ := strconv.Atoi(numStr)`;
+    # default is false: such cases aren't reported by default.
+    check-blank: false
+
+    # [deprecated] comma-separated list of pairs of the form pkg:regex
+    # the regex is used to ignore names within pkg. (default "fmt:.*").
+    # see https://github.com/kisielk/errcheck#the-deprecated-method for details
+    ignore: fmt:.*,io/ioutil:^Read.*
+
+  govet:
+    # report about shadowed variables
+    check-shadowing: false
+
+  revive:
+    # confidence for issues, default is 0.8
+    confidence: 0.8
```

Comment:
I suggest removing this block. Doing so causes `revive` to catch the things that `golint` used to catch. See https://github.com/crossplane/crossplane/pull/3462/commits/74e387bcecae72faee5280888e7ef216a06a498c for more context.

Reply:
Thank you @negz, very helpful. When this configuration block is removed, we now have to document, for example, exported API. Removed this block, and added some package docs and exported API docs.